snuba/state.py

```diff
@@ -272,11 +275,21 @@ def get_config_changes():
 
 # Query Recording
 
+
+def safe_dumps_default(value: Any) -> Any:
+    if isinstance(value, Mapping):
```

Comment:
why not just returning value itself if it is an instance of dict ? So we do not reallocate another dictionary when the input was already safe.

Reply:
The `json` encoder expects an actual dictionary instance, not something that conforms to the `Mapping` interface.

Reply:
right I mean
```
    if isinstance(value, dict):
        return value
    if isinstance(value, Mapping):
        return {**value}
```

Reply:
That first conditional won't ever be hit:

```
ted@veneno % python3
Python 3.7.2 (default, Feb 12 2019, 08:16:11) 
[Clang 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> def default(value):
...   assert False
... 
>>> json.dumps({}, default=default)
'{}'
```

Reply:
sorry, did not see the partial.

Reply:
No worries, it was a good question.